enhancements/sig-architecture/70-spread-policy/README.md

```diff
@@ -0,0 +1,288 @@
+
+### Use Case 5: Joint Spread
+
+To express hierarchical topology structure, clusters are labeled with multiple topology keys (i.e., region, zone, etc.), we want them to work jointly to determine the spread decision.
+
+**Example.** We want four clusters from aws to put our backend service.
+Further, we want to spread the workload first across regions and then across zones.
+Assuming all zones here can provide enough clusters.
+Then, the scheduling result should contain one cluster from us-east-1a, one cluster from us-east-1b, and other two clusters from us-west-1a.
+
+### Use Case 6: Spread with Ratio (Out of Scope)
+
+Sometimes, we may need to spread the workload with a specified ratio.
+Let us consider a scenario where most of the access to the workload is from the east coast of the US, even spread between us-east-1 and us-west-1 would be inappropriate.
+Instead, we want to place more workload on us-east-1 than us-west-1.
+Besides, we want to control such preferences by user-specified configurations.
+
+**Example.** We want six clusters to deploy our backend service.
+Moreover, we want 2/3 of them to be located in us-east-1 and 1/3 to be located in us-west-1 since there are more clients from the east coast of the US.
+
+## Goal
+Generally, we have two goals for the spread policy:
+1. Cover the full functionality in the Motivation section;
+2. Implement a high-performance and scalable mechanism that works for thousands of clusters efficiently.
+
+## Proposal
+
+### API Spec
+```go
+type PlacementSpec struct {
+	// ...
+
+	// SpreadPolicy defines how placement decisions should be distributed among a
+	// set of clusters.
+	// +optional
+	SpreadPolicy SpreadPolicy `json:"spreadPolicy,omitempty"`
+}
+
+// SpreadPolicy defines how placement decisions should be spreaded among the
+// clusters.
+type SpreadPolicy struct {
+	// SpreadConstraints defines how placement decision should be distributed among a
+	// set of clusters.
+	// +optional
+	// +kubebuilder:validation:MaxItems=64
+	SpreadConstraints []SpreadConstraintsTerm `json:"spreadConstraints,omitempty"`
+}
+
+// SpreadConstraintsTerm defines a terminology to spread placement decisions
+type SpreadConstraintsTerm struct {
+	// TopologyKey is either a label key or a cluster claim name of ManagedClusters
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]$`
+	// +kubebuilder:validation:MaxLength=316
+	TopologyKey string `json:"topologyKey"`
+
+	// TopologyKeyType indicates the type of TopologyKey. It could be Label or Claim.
+	// +required
+	// +kubebuilder:validation:Required
+	TopologyKeyType TopologyKeyType `json:"topologyKeyType"`
+
+	// MaxSkew describes the degree to which the workload may be unevenly distributed.
+	// If not omitted, the scheduler would not schedule if MaxSkew cannot be satisfied.
```

Comment:
If present, the scheduler will not schedule if the MaxSkew cannot be satisfied.

Reply:
Done :)